### ISSUE TITLE
portable-ruby@2.2 links to Homebrew libraries on macOS

### ISSUE THREAD

**Issue by xu-cheng** [Homebrew/homebrew-portable#31]
```
$ brew linkage portable-ruby@2.2
System libraries:
  /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
  /usr/lib/libSystem.B.dylib
  /usr/lib/libncurses.5.4.dylib
  /usr/lib/libobjc.A.dylib
  /usr/lib/libutil.dylib
  /usr/lib/libz.1.dylib
Homebrew libraries:
  /usr/local/opt/libyaml/lib/libyaml-0.2.dylib (libyaml)
Possible undeclared dependencies:
  libyaml
```

As shown above, portable-ruby@2.2 will accidentally link to Homebrew library (libyaml) instead of portable-yaml. Test is done on macOS 10.12.

cc @mistydemeo

**Comment by xu-cheng** [comment]
Test shows this is caused by https://github.com/Homebrew/homebrew-portable/commit/6ce1fb3042ddee6caa43ec6372c296fc3375a857.

I wonder whether the cause is because `make` method bypass superenv?

**Comment by mistydemeo** [comment]
Good catch - I missed this! Tiger doesn't have superenv yet, so I didn't notice it in that environment.

This should probably be set to just run `make` or `gmake`, not the full path, and as long as the Homebrew-installed make is before `/usr/bin` in the PATH, it should be fine.

**Comment by mistydemeo** [comment]
Fixed via Homebrew/brew#2883.
